gatsby-node.js

```diff
@@ -62,75 +64,146 @@ exports.createPages = async (createPagesArgs) => {
 const createCareerPages = async ({ actions }) => {
   const { createPage } = actions;
 
-  const jobsXmlResponse = await fetch(PERSONIO_JOBS_URL);
-  const jobsXml = await jobsXmlResponse.text();
-  const jobsParse = xmlParser.parse(jobsXml);
-  const positions = jobsParse['workzag-jobs'].position;
+  const positionsGroupedByLanguage = LANGUAGES.map(async (langKey) => {
+    const PERSONIO_JOBS_URL_LANG = PERSONIO_JOBS_URL.concat(
+      `?language=${langKey}`,
+    );
+    const jobsXmlResponse = await fetch(PERSONIO_JOBS_URL_LANG);
+    const jobsXml = await jobsXmlResponse.text();
+    const options = {
+      tagValueProcessor: (a) => decode(a), // &#039; -> '
+    };
+    const jobsParse = xmlParser.parse(jobsXml, options);
+    const positions = jobsParse['workzag-jobs'].position;
+    return { positions, langKey };
+  });
 
-  positions.forEach((position) => {
-    position.satellytesPath = `/career/${position.id}-${slugify(position.name, {
-      lower: true,
-    })}/`;
-
-    // we need to normalize the jobDescription to an array, otherwise gatsby
-    // complains about different types for the same variable
-    position.jobDescriptions.jobDescription = Array.isArray(
-      position.jobDescriptions.jobDescription,
-    )
-      ? position.jobDescriptions.jobDescription
-      : [position.jobDescriptions.jobDescription];
-
-    position.jobDescriptions.jobDescription =
-      position.jobDescriptions.jobDescription.map((description) => {
-        return {
-          name: description.name.trim(),
-          value: description.value.trim(),
-        };
-      });
+  await Promise.all(positionsGroupedByLanguage).then(async (values) => {
+    for (let i = 0; i < values.length; i++) {
+      const positions = values[i].positions;
+      const langKey = values[i].langKey;
+
+      // position is valid if it has a job description and slug
+      const isValidPosition = (position) => {
+        return (
+          position.jobDescriptions.jobDescription &&
```

Comment:
This check is only about the `jobDescription`, this is why I would extract the value first and then check it.

```js
const { jobDescription } = position.jobDescriptions
return (...)
```